diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt

```diff
@@ -207,20 +200,27 @@ class CommentsFormatting(private val configRules: List<RulesConfig>) : Rule("kdo
     }
 
     private fun checkSpaceBeforeComment(node: ASTNode, configuration: CommentsFormattingConfiguration) {
-        if (node.treeParent.firstChildNode != node) {
-            if (!node.treePrev.isWhiteSpace()) {
-                // if comment is like this: val a = 5// Comment
-                COMMENT_WHITE_SPACE.warnAndFix(configRules, emitWarn, isFixMode,
-                        "There should be ${configuration.maxSpacesBeforeComment} space(s) before comment text, but are none in ${node.text}", node.startOffset, node) {
-                    node.treeParent.addChild(PsiWhiteSpaceImpl(" ".repeat(configuration.maxSpacesBeforeComment)), node)
-                }
-            } else if (!node.treePrev.textContains('\n') && node.treePrev.text.length != configuration.maxSpacesBeforeComment) {
-                // if there are too many spaces before comment
+        if (node.treeParent.firstChildNode == node)
+            return
+        if (node.treeParent.elementType == FILE) {
+            if (node.treePrev.isWhiteSpace()) {
```

Comment:
Also please add comment here, that this case is for top-level comments that are located in the beginning of the line and therefore don't need any spaces before.